lightning/src/offers/invoice_request.rs

```diff
@@ -2981,4 +3010,31 @@ mod tests {
 			Err(_) => panic!("unexpected error"),
 		}
 	}
+
+	#[test]
+	fn test_string_truncate_safe() {
+		// We'll correctly truncate to the nearest UTF-8 code point boundary:
+		// ❤      variation-selector
+		// e29da4 efb88f
+		let s = String::from("❤️");
+		for idx in 0..(s.len() + 5) {
+			if idx >= s.len() {
+				assert_eq!(s, string_truncate_safe(s.clone(), idx));
+			} else if (3..s.len()).contains(&idx) {
+				assert_eq!("❤", string_truncate_safe(s.clone(), idx));
+			} else {
+				assert_eq!("", string_truncate_safe(s.clone(), idx));
```

Comment:
Maybe just listing all the cases without a loop is easier here because there are only a few?

Reply:
That's 5 cases then? I'd think it would be fewer lines of code than what you have currently. As a reviewer, I had to look twice to see how this works, and thought it could be better with all of it written out.